test/datatype.test.js

```diff
@@ -35,6 +35,16 @@ describe('datatypes', function() {
     db.automigrate(['Model'], done);
   });
 
+  it('should resolve "type" sub-property', function() {
+    const Account = db.define('Account', {
+      item: {type: {
+        itemname: {type: String},
+        type: {type: String},
+      }},
+    });
+    Account.definition.properties.item.type.should.not.equal(String);
```

Comment:
Can we add a test case to cover the following?
```js
const Account = db.define('Account', {
      type: String, // account type - checking/saving
      id: String
    });
```

Reply:
Sure, thanks.